scripts/data_prep/convert_text_to_mds.py

```diff
@@ -0,0 +1,352 @@
+    input_folder: str,
+    concat_tokens: int,
+    eos_text: str,
+    bos_text: str,
+    no_wrap: bool,
+    max_mds_writer_workers: int,
+    compression: str,
+    processes: int,
+    args_str: str,
+    reprocess: bool,
+):
+    done_file_name = get_done_file_name()
+    is_remote_output = is_remote_path(output_folder)
+
+    object_names = get_object_names(input_folder)
+
+    # Check if the text files in the bucket have already been processed.
+    if not reprocess and is_already_processed(output_folder, done_file_name,
+                                              args_str, object_names):
+        print(
+            f'Input folder {input_folder} is already processed at {output_folder} and reprocess is set to False. Set reprocess to True if you would like to force reprocessing.'
+        )
+        return
+
+    # Use a temporary local directory if the output is remote and there are more than 1 processes
+    local_output_folder = tempfile.TemporaryDirectory(
+    ).name if is_remote_output else output_folder
+
+    if processes > 1:
+        # Download and convert the text files in parallel
+        args = get_task_args(object_names, local_output_folder, input_folder,
+                             processes, tokenizer_name, concat_tokens, eos_text,
+                             bos_text, no_wrap, compression,
+                             max_mds_writer_workers)
+        with ProcessPoolExecutor(max_workers=processes) as executor:
+            list(executor.map(download_and_convert_starargs, args))
+
+        # Merge the mds shards from each of the processes into a single folder
+        merge_shard_groups(local_output_folder)
+    else:
+        download_and_convert(object_names, local_output_folder, input_folder,
+                             tokenizer_name, concat_tokens, eos_text, bos_text,
+                             no_wrap, compression, max_mds_writer_workers)
+
+    # Write a done file with the args and object names
+    write_done_file(local_output_folder, done_file_name, args_str, object_names)
+
+    if is_remote_output:
+        # Upload the local output to the remote location
+        output_object_store = cast(
+            ObjectStore, maybe_create_object_store_from_uri(output_folder))
+        _, _, output_folder_prefix = parse_uri(output_folder)
+        files_to_upload = os.listdir(local_output_folder)
+
+        for file in files_to_upload:
+            assert not os.path.isdir(file)
+            remote_path = os.path.join(output_folder_prefix, file)
+            output_object_store.upload_object(
+                remote_path, os.path.join(local_output_folder, file))
+
+
+def _args_str(original_args: Namespace) -> str:
+    """Create a string from the args to determine whether to reprocess.
+
+    Args:
+        original_args (Namespace): args to transform
+    """
+    args = copy.deepcopy(original_args)
+
+    # Remove args that do not affect the final result.
+    delattr(args, 'max_mds_writer_workers')
```

Comment:
even though i think this is safe given the flow of this script, can we make this function non destructive? seems unnecessary